### ISSUE TITLE
Please create a way for the autostart to be se to not final and we can change it to false or true.

### ISSUE THREAD

**Issue by kate-abracosa** [juliansteenbakker/mobile_scanner#1011]
The reason is the permission setting you made on the package was not flawless. if camera is not permitted then nothing already it cannot be read also externally the status of the permission when it was permanently denied. So just make the autostart to be customizable so on initstate i can make my own getpermission then when the app was set to paused state it wont turns white since i will set it to autstart = true if permission is granted

cameraController = MobileScannerController(
        facing: CameraFacing.back,
        autoStart: false,
    );
    getPermission();

**Comment by diegonc** [comment]
Can't you just call the `start()` methond on `cameraController` according to the `getPermission()` result?

**Comment by kate-abracosa** [comment]
Yes, I called cameracontroller.start() if permission is granted. However the problem was if the app state is set to paused or inactive. I am seeing white and the scanner is not working. This is because i set the autostart to false. If I set it to true. I wont have an issue even if the app is set to paused or inactive. However, if i did not set the autostart to false, I will get an issue with permission setting since on initstate the mobilescanner has its own permission but if the permission. the permission_handler which is another package does not seem to recognize the event so I cannot show a bottomsheet which says We need the permission to access your camera allow (open appsetting) or  later and close the full screen

**Comment by diegonc** [comment]
Yeah, I also faced that issue. I copied some bits of the old `mobile_scanner` version (the one that used to have `autoStart`)
Something like the code below should work. As you can see, I'm using the `MobileScanner` widget from a `StatefulWidget` but I guess the overall approach should work in other setups.

```dart
class _ScannerState extends State<Scanner> with WidgetsBindingObserver {
  MobileScannerController? _controller;
  bool _resumeFromBackground = false;

  @override
  void onInit() {
    super.onInit();
    _controller = MobileScannerController(
        detectionSpeed: DetectionSpeed.normal, autoStart: false);
     WidgetsBinding.instance.addObserver(this);
  }

  @override
  void dispose() {
    _controller?.dispose();
    WidgetsBinding.instance.removeObserver(this);
    super.dispose();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    if (_controller != null && _controller!.isStarting) {
      return;
    }

    switch (state) {
      case AppLifecycleState.resumed:
        if (_resumeFromBackground) {
          _controller.start();
        }
      case AppLifecycleState.inactive:
        _resumeFromBackground = widget.enabled.isTrue;
        _controller.stop();
      default:
        break;
    }
  }
}
```

**Comment by kate-abracosa** [comment]
Hi Diego, 

Thank you so much, I also did the same yesterday on my code and it worked.

**Comment by navaronbracke** [comment]
> Can't you just call the `start()` method on `cameraController` according to the `getPermission()` result?
> 
> EDIT: BTW I just noticed the beta for verison 5.0.0 removed the autoStart parameter, so the issue is probably no longer relevant

I do not understand the original intention here? Indeed, we removed the `autoStart` parameter, because doing lifecycle things inside the `MobileScanner` widget caused issues with widget tree reparenting.
